src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/factory/enrollment/EnrollmentFactory.js

```diff
@@ -310,7 +310,7 @@ export class EnrollmentFactory {
                     });
                 } else if (cachedProgramSections) {
                     // $FlowFixMe
-                    cachedProgramSections.asyncForEach(async (programSection) => {
+                    await cachedProgramSections.asyncForEach(async (programSection) => {
```

Comment:
The fix looks good @eirikhaugstulen!
Do you think it makes sense to also add await in the other two places https://github.com/EyeSeeTea/capture-app/pull/2#discussion_r2014150265 (EnrollmentFactory.js#L269 and TeiRegistrationFactory.js#L105)?
Thanks!

Reply:
Yes, we should add `await` for those two as well (I double checked that those are the only other places in the code where we call `asyncForEach` without `await`)